extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -676,107 +670,113 @@ export class AmpA4A extends AMP.BaseElement {
+            'frameborder': '0', 'allowfullscreen': '',
+            'allowtransparency': '', 'scrolling': 'no'}));
+      this.applyFillContent(iframe);
+      const fontsArray = [];
+      if (creativeMetaData.customStylesheets) {
+        creativeMetaData.customStylesheets.forEach(s => {
+          const href = s['href'];
+          if (href) {
+            fontsArray.push(href);
           }
-          return installFriendlyIframeEmbed(
-            iframe, this.element, {
-              url: this.adUrl_,
-              html: creativeMetaData.minifiedCreative,
-              extensionIds: creativeMetaData.customElementExtensions || [],
-              fonts: fontsArray,
-            }, embedWin => {
-              installUrlReplacementsForEmbed(this.getAmpDoc(), embedWin,
-                new A4AVariableSource(this.getAmpDoc(), embedWin));
-            }).then(friendlyIframeEmbed => {
-              // Ensure visibility hidden has been removed (set by boilerplate).
-              const frameDoc = friendlyIframeEmbed.iframe.contentDocument ||
-                friendlyIframeEmbed.win.document;
-              setStyle(frameDoc.body, 'visibility', 'visible');
-              // Capture phase click handlers on the ad.
-              this.registerExpandUrlParams_(friendlyIframeEmbed.win);
-              // Bubble phase click handlers on the ad.
-              this.registerAlpHandler_(friendlyIframeEmbed.win);
-              this.rendered_ = true;
-              this.onAmpCreativeRender();
-              return true;
-            });
-        } catch (e) {
-          dev().error('AMP-A4A', 'Error injecting creative in friendly frame',
-              e);
-          // If we fail on any of the steps of Shadow DOM construction, just
-          // render in iframe.
-          // TODO: report!
-          return false;
-        }
+        });
       }
-    });
+      return installFriendlyIframeEmbed(
+        iframe, this.element, {
+          url: this.adUrl_,
+          html: creativeMetaData.minifiedCreative,
+          extensionIds: creativeMetaData.customElementExtensions || [],
+          fonts: fontsArray,
+        }, embedWin => {
+          installUrlReplacementsForEmbed(this.getAmpDoc(), embedWin,
+            new A4AVariableSource(this.getAmpDoc(), embedWin));
+        }).then(friendlyIframeEmbed => {
+          // Ensure visibility hidden has been removed (set by boilerplate).
+          const frameDoc = friendlyIframeEmbed.iframe.contentDocument ||
+            friendlyIframeEmbed.win.document;
+          setStyle(frameDoc.body, 'visibility', 'visible');
+          // Capture phase click handlers on the ad.
+          this.registerExpandUrlParams_(friendlyIframeEmbed.win);
+          // Bubble phase click handlers on the ad.
+          this.registerAlpHandler_(friendlyIframeEmbed.win);
+          this.onAmpCreativeRender();
+          return true;
+        });
+    } catch (e) {
```

Comment:
this will not catch any error thrown inside the promise then blocks.

Yuu'll want to use `the().catch()`.